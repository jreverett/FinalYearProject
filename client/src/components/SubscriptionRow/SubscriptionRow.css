.button-cancel-subscription {
  background-color: transparent;
  border: none;
}

.subscription-row-icon {
  color: var(--red);
  margin-bottom: 5px;
}
