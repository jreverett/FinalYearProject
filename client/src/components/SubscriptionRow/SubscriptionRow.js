import React, { Component } from 'react';
import { MdDeleteForever } from 'react-icons/md';
import { FaCalendarTimes } from 'react-icons/fa';
import { toast } from 'react-toastify';
import { eventService, userService } from '../../services';
import { formatDateTime } from '../../utilities';
import './SubscriptionRow.css';

class SubscriptionRow extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      start: '',
      removed: false
    };
  }

  componentDidMount() {
    eventService.getEvent(this.props.eventID).then(event => {
      event = event.data;
      this.setState({ title: event.title, start: event.start });
    });
  }

  unsubscribe = () => {
    userService.unsubscribe(this.props.userID, this.props.eventID).then(
      () => {
        this.setState({ removed: true });
        toast.success(
          <p>
            <FaCalendarTimes className="form-icon" />
            Unsubscribed from {this.state.title}
          </p>
        );
      },
      error => {
        toast.error(error);
      }
    );
  };

  render() {
    const { title, start, removed } = this.state;
    return (
      <div
        className={`profile-tab-item-container ${removed ? 'hidden' : null}`}
      >
        <div className="profile-tab-item-text-container">
          <p>{`${title} @ ${formatDateTime(start)}`}</p>
        </div>
        <div>
          <button
            className="button-cancel-subscription"
            onClick={this.unsubscribe}
          >
            <MdDeleteForever className="subscription-row-icon" size={'1.5em'} />
          </button>
        </div>
      </div>
    );
  }
}

export default SubscriptionRow;
