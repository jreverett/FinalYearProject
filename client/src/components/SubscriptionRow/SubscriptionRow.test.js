import React from 'react';
import renderer from 'react-test-renderer';
import SubscriptionRow from './SubscriptionRow';

describe('<SubscriptionRow />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<SubscriptionRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
