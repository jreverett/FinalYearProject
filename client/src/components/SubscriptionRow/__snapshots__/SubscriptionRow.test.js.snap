// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionRow /> matches the snapshot 1`] = `
<div
  className="table-item-container null"
>
  <div
    className="profile-tab-item-text-container"
  >
    <p>
      loading...
    </p>
  </div>
  <div>
    <button
      className="icon-button"
      onClick={[Function]}
    >
      <svg
        className="row-delete-icon"
        fill="currentColor"
        height="1.5em"
        size="1.5em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </button>
  </div>
</div>
`;
